src/renderer/plugins/SyncWindowRectPlugin.ts

```diff
@@ -1,25 +1,28 @@
-import { RendererApp, RendererAppPlugin } from "../RendererApp"
-import { answerMain, callMain } from "../RendererIPC"
+import { RendererApp } from "../RendererApp"
+import { RendererEnvironment } from "../RendererEnvironment"
+import { RendererIPCPeer } from "../RendererIPC"
 import { RendererState } from "../RendererState"
 
-export const SyncWindowRectPlugin: RendererAppPlugin = (app) => {
-	return new SyncWindowRectController(app)
-}
-
 /**
  * Updates the local state to match the main process state.
  */
-class SyncWindowRectController {
+export class SyncWindowRectPlugin {
+	private ipc: RendererIPCPeer
+	private app: RendererApp
 	private listeners: (() => void)[] = []
 
-	constructor(private app: RendererApp) {
+	constructor(environment: RendererEnvironment) {
+		const { ipc, app } = environment
+		this.ipc = ipc
+		this.app = app
+
 		this.listeners.push(
-			answerMain.updatePosition(({ x, y }) => {
+			ipc.answer.updatePosition(({ x, y }) => {
 				app.dispatch.moveWindow({ x, y })
 			})
 		)
 		this.listeners.push(
-			answerMain.updateSize(({ height, width }) => {
+			ipc.answer.updateSize(({ height, width }) => {
 				app.dispatch.resizeWindow({ height, width })
 			})
 		)
```

Comment:
Fyi, I've had this lead to an infinite bug, where the window spazzed out from calls being sent back and forth from the renderer to the main. It only happens occasionally though. It ends up in this state where main calls to renderer and renderer calls back to main...you can follow this logic here, it answers to main's `updatePosition`, updates its own state, and then calls back to main in the effect